spinnaker-monitoring-third-party/third_party/prometheus/prometheus.conf

```diff
@@ -1,6 +1,6 @@
 start on filesystem or runlevel [2345]
 
-exec /opt/PROMETHEUS_VERSION/prometheus \
-  -config.file /opt/PROMETHEUS_VERSION/spinnaker-prometheus.yml \
-  -storage.local.path /opt/PROMETHEUS_VERSION/data \
+exec /opt/prometheus/prometheus \
```

Comment:
I'm not sure how to handle this. For practical purposes, the problem here is in the data because the old data store (in the versioned directory) is going to be abandoned and a new version-independent one will be used. Going forward, this would be preserved across versions.

As for your question on configuration, the configuration is local to the versioned directory. This introduces a new version so is going to abandon the old config and write new one. To preserve old changes, you'd need to go back in and edit/carry forward for the time being.

I could have some kind of option indicating to keep the old version or replace it. To keep it, for this version, I could look for the one known previous version. In the future I could follow the symbolic link before replacing it. I'd copy it forward. Except it might not work or at least new features might not be properly configured. To be really fancy, which I'm not planning on, I suppose you'd need to some kind of 3-way merge between the old original, users changes to it, and new update.